src/main/java/life/mosu/mosuserver/applicaiton/faq/FaqService.java

```diff
@@ -0,0 +1,113 @@
+package life.mosu.mosuserver.applicaiton.faq;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import life.mosu.mosuserver.domain.faq.FaqAttachmentJpaEntity;
+import life.mosu.mosuserver.domain.faq.FaqAttachmentRepository;
+import life.mosu.mosuserver.domain.faq.FaqJpaEntity;
+import life.mosu.mosuserver.domain.faq.FaqRepository;
+import life.mosu.mosuserver.global.exception.CustomRuntimeException;
+import life.mosu.mosuserver.global.exception.ErrorCode;
+import life.mosu.mosuserver.infra.storage.application.S3Service;
+import life.mosu.mosuserver.infra.storage.domain.Folder;
+import life.mosu.mosuserver.presentation.faq.dto.FaqCreateRequest;
+import life.mosu.mosuserver.presentation.faq.dto.FaqResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
```

Comment:
`readOnly = true` 옵션도 좋지만, 이는 다른 것과 합쳐서 상호작용하게 된다면, 트랜잭션이 많이 생기는 장애를 발생할 수도 있어요!
```suggestion
@Transactional(readOnly = true, propagation = SUPPORTS)
```
으로 구성할 경우, 전파옵션에 따라서 트랜잭션이 없으면 따로 실행되지 않고, 트랜잭션이 존재하면 하나의 트랜잭션으로 실행됩니다.

오히려, 읽기는 트랜잭션 처리가 없는 경우에 더 빠른 효율이 납니다.

https://tech.kakaopay.com/post/jpa-transactional-bri/

트랜잭션을 유지해야하는 타당성이 없다면 해당 전파옵션을 고려해보세요!🌟

Reply:
아하! 트랜잭션의 사이드 이펙트에 대해서 자세히 알지는 못했는데, 이번 기회에 알게되었습니다! 감사합니다 ☺️